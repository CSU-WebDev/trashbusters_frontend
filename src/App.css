/* #root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
} */

body, html, #root, .App{
  height: 100%;
}

/*:root {
  --dark-mode-gray: rgb(64, 64 ,64);
  --light-mode-offwhite: #56585a4d;
  --background-offwhite: rgb(240, 230, 230);
}*/

#navigation-bar {
  color: white;
  background-color: #658E9C;
  font-family: "Lucida Bright", Georgia, serif;
}

.navbar-brand {
  margin-left: 50px;
  font-size: xx-large;
  color: white !important;
}

.navbar-links {
  margin-right: 200px;
  font-size: xx-large;
}

.navbar-link {
  color: white;
  transition: all 0.1s ease-in-out;
}

.navbar-link:hover, .navbar-link.active {
  color: white !important;
  text-shadow: 0 0 10px gray, 0 0 20px rgb(87, 87, 87), 0 0 30px rgb(34, 34, 34);
  transform: scale(1.1);
}

#brand {
  pointer-events: none;
}

.wrapper{
  background-color: #004D21;
}

.box{
  background-color: rgb(136, 123, 10);
  padding: 15px;
  border-radius: 25px;
  font-family: "Lucida Bright", Georgia, serif;
  color: white;
  box-shadow: 1px 1px 10px 10px rgba(5, 5, 5, 0.466);
}

.center {
  display: flex;
  width: 100%;
  max-width: 60%;
  margin: 0 auto;
  justify-content: center;
}

.height100{
  height: 100%;
}

.row-space{
  margin-top: 15px;
  margin-bottom: 15px;
}

#footer {
  background-color: #658E9C;
  font-family: "Lucida Bright", Georgia, serif;
  color: white;
  padding-top: 10px;
}

.center-component{
  align-items: center;
  justify-content: center;
  display: flex;
}

#grid-container {
  padding-bottom: 15px;
}