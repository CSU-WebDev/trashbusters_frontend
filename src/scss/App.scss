@import 'variables';
@import 'navigationbar';
@import 'main';
@import 'footer';