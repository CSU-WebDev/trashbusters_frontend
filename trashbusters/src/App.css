/* #root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
} */

body, html, #root, .App{
  height: 100%;
}

:root {
  --dark-mode-gray: rgb(64, 64 ,64);
  --light-mode-offwhite: #56585a4d;
  --background-offwhite: rgb(240, 230, 230);
}

#navigation-bar {
  color: white;
  background-color: gray;
}

.navbar-brand {
  margin-left: 50px !important;
  font-size: xx-large;
  color: white;
}

.navbar-links {
  margin-right: 200px !important;
  font-size: xx-large;
}

.navbar-link {
  color: white;
  transition: all 0.1s ease-in-out;
}

.navbar-link:hover, .navbar-link.active {
  color: white !important;
  text-shadow: 0 0 10px gray, 0 0 20px gray, 0 0 30px gray;
  transform: scale(1.1);
}

#brand {
  pointer-events: none;
}

.wrapper{
  background-color: rgb(17, 80, 17);
  height: 100%;
}

.box{
  background-color: burlywood;
  padding: 15px;
  border-radius: 5px;
}

.center {
  display: flex;
  width: 100%;
  max-width: 60%;
  margin: 0 auto;
  justify-content: center;
}

.height100{
  height: 100%;
}

.row-space{
  margin-top: 15px;
  margin-bottom: 15px;
}