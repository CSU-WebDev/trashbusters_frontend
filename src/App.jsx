// import { useState } from 'react'
import './App.css'
import NavigationBar from './components/Navbar.jsx'
import Main from './components/Main.jsx'
import Footer from './components/Footer.jsx'

function App() {

  return (
    <div className="App">
      <NavigationBar />
      <Main />
      <Footer />
    </div>
  )
}

export default App
