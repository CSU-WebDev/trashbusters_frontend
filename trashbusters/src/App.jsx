import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import NavigationBar from './components/Navbar'
import Main from './components/Main'

function App() {

  return (
    <div className="App">
      <NavigationBar />
      <Main />
    </div>
  )
}

export default App
