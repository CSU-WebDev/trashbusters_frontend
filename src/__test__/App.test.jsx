import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../App';

describe('App component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
