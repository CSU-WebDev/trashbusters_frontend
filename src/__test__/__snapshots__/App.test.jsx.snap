// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="sticky-top navbar navbar-expand-md navbar-light"
      id="navigation-bar"
    >
      <span
        class="navbar-brand"
        id="brand"
      >
        Trashbusters
      </span>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="justify-content-end navbar-links navbar-nav"
        >
          <a
            class="navbar-link nav-link"
            data-rr-ui-event-key="#about-container"
            href="#about-container"
          >
            About
          </a>
          <a
            class="navbar-link nav-link"
            data-rr-ui-event-key="#trashmap-container"
            href="#trashmap-container"
            id="trashmap-link"
          >
            Trash-Map
          </a>
          <a
            class="navbar-link nav-link"
            data-rr-ui-event-key="#contact-container"
            href="#contact-container"
          >
            Contact
          </a>
        </div>
      </div>
    </nav>
    <div
      class="container-fluid"
      data-testid="main"
      id="main"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="mt-5 box container-fluid"
            data-testid="pitch-container"
            id="pitch-container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h3
                  class="text-center"
                >
                  Which do you prefer???
                </h3>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="mt-4 mb-4 container"
                  data-testid="pics-container"
                  id="pics-container"
                >
                  <div
                    class="justify-content-center row"
                    data-testid="pics-row"
                  >
                    <div
                      class="align-items-center col-md-6"
                      data-testid="pics-col-1"
                      s="12"
                    >
                      <img
                        class="pic img-fluid"
                        data-testid="pic"
                        src="litter-before.jpg"
                      />
                    </div>
                    <div
                      class="align-items-center col-md-6"
                      data-testid="pics-col-2"
                      s="12"
                    >
                      <img
                        class="pic img-fluid"
                        data-testid="pic"
                        src="litter-after.jpg"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="box container-fluid"
            data-testid="about-container"
            id="about-container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h3
                  class="text-center"
                >
                  About
                </h3>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <p
                  data-testid="about-paragraph"
                  id="about-paragraph"
                >
                  Litter is 
                  <i>
                    never
                  </i>
                   fun to look at. Join the fight and help bust some trash! Below is our trash map. You can use it to mark areas on the map where you have spotted litter. Be sure to add a description if you drop a new pin! You can also delete pins to cleanup sites once all trash has been removed. Thank you for making our community more beautiful!
                </p>
              </div>
            </div>
          </div>
          <div
            class="box container"
            data-testid="trashmap-container"
            id="trashmap-container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h3
                  class="text-center"
                >
                  Trash Map
                </h3>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
                id="trashmap-container-col"
              >
                <div
                  id="trashmap"
                >
                  <div />
                  <div>
                    Loading...
                  </div>
                  <div
                    id="spinnerDiv"
                  >
                    <div
                      class="spinner-border text-dark"
                      id="spinner"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="box container"
            data-testid="contact-container"
            id="contact-container"
          >
            <div
              class="row"
            >
              <h3
                class="text-center"
              >
                Contact Us
              </h3>
            </div>
            <div
              class="row"
            >
              <div
                class="d-flex justify-content-center col"
              >
                <form
                  action="https://formspree.io/f/xyyaglza"
                  class=""
                  data-testid="form"
                  id="form"
                  method="POST"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="mb-2 form-label"
                      for="formBasicEmail"
                    >
                      Email address
                    </label>
                    <input
                      class="form-control"
                      id="formBasicEmail"
                      name="email"
                      placeholder="Enter email"
                      required=""
                      type="email"
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="mb-2 form-label"
                      for="formBasicName"
                    >
                      Name
                    </label>
                    <input
                      class="form-control"
                      data-testid="name-input"
                      id="formBasicName"
                      name="name"
                      placeholder="Enter your name"
                      required=""
                      type="text"
                    />
                  </div>
                  <div
                    class="mb-3"
                  >
                    <label
                      class="mb-2 form-label"
                      for="formBasicMessage"
                    >
                      Message
                    </label>
                    <input
                      class="form-control"
                      data-testid="message-input"
                      id="formBasicMessage"
                      name="message"
                      placeholder="Enter your message"
                      required=""
                      rows="3"
                      type="textarea"
                    />
                  </div>
                  <button
                    class="btn-lg btn btn-primary"
                    type="submit"
                  >
                    Submit
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      data-testid="footer"
      id="footer"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <p>
              Trashbusters Â© May 2023 
            </p>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
